activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -0,0 +1,403 @@
+# frozen_string_literal: true
+
+begin
+  require "redis"
+  require "redis/distributed"
+rescue LoadError
+  warn "The Redis cache store requires the redis gem. Please add it to your Gemfile."
```

Comment:
Should we be doing `gem 'redis` in this file so we assert the minimum version requirement?

Reply:
Yes. I haven't carefully reviewed the oldest supported version yet. Will probably be Redis 3.3.0 with support for connect/read/write_timeout.

Reply:
Actually, requiring latest redis-rb (4.0.1+) is appropriate for a new feature like this. Then sharded cache `read_multi` will just work without needing additional warnings.